### ISSUE TITLE
Gan model training, losses and epoch

### ISSUE THREAD

**Issue by Kirin-kun** [deepfakes/faceswap-playground#118]
I'm currently training a dataset with the GAN128 model of the faceswap repo and I'd like to know how to determine when training has reached a "good enough" threshold.

I've read in other discussions, that you'd usually go for 10k iterations just training the faces, then 10k with perceptual loss on, then, after 15k, you'd turn on mask refinement.

And then, here is my question : which number should I look at? The display shows:

[time] [xx/num_epochs][xxx] Loss_DA: Loss_DB: Loss_GA: Loss_GB:

I have a batch size of 10 (can't go any higher, with just 6GB of VRAM).

Also, the time: what use is it? 

I know what time it is. It's also written in the right bottom corner of my screen, on the clock, thank you.

Then, when talking about iterations, are we talking about the first, or the second number (xx or xxx)? I guess the xx, because the other value gets high quickly without the preview really improving. 

Also, would it be possible to save somewhere the number of iterations the model has already been running? Because I can't let my computer run 24/24 and each time I stop the training, I don't want to bother to write down what the value was and so, I'm lost as to how far in the iteration count I am

Finally, what kind of values in the losses should I look forward to? Low loss for both the discriminators? What about the generators? Actually, with a few hundred faces in the A and B set, the losses don't really go down, or really slowly. And after a few thousand iterations (in xx value), the generated faces in the masked preview are still looking twisted and not realistic. 

I think I still have a long way to go to get something acceptable, but I don't want to train for days and realize that there won't be any result anyway. I'm really curious about this model, because I want to see how it handles occlusion. But after 3 days of training, it doesn't look like close to completion.

**Comment by hoomanNo5** [comment]
FWIW I still haven't had close to the similar results that Clorr got here: https://github.com/deepfakes/faceswap/issues/247

training on a 1080 using this after some experimentation: 
python faceswap.py train -A [many faces] -B [one face] -t GAN128 -pl -ag -bs 32

Trying to achieve a many-to-one face swap:

- 13k "many faces" as A input
- 13k "one face" as B input
[107/num_epochs][41105] Loss_DA: 0.016701 Loss_DB: 0.009724 Loss_GA: 0.965387 Loss_GB: 1.166446  

I believe the DA DB loss is pretty good. I...think the G loss is acceptable. I know GANs can be a bitch to train but this is pretty frustrating.

**Comment by Kirin-kun** [comment]
No answer, so I guess I'll close it.
